c10/util/Logging.h

```diff
@@ -309,17 +309,45 @@ C10_API void LogAPIUsage(const std::string& context);
 // fields, will add more fields as follow ups such as performance stats,
 // internal states and env variables and etc.
 struct DDPLoggingData {
-  // Data that can be got during DistributedDataParallel construction time
-  int world_size;
-  int rank;
-  std::string module_name;
-  std::vector<int> device_ids;
-  int output_device;
-  bool broadcast_buffers;
-  int bucket_cap_mb;
-  bool find_unused_parameters;
-  bool gradient_as_bucket_view;
-  std::string backend_name;
+// Data that can be got during DistributedDataParallel construction time
+  int world_size = -1;
+  int rank = -1;
+  std::string module_name = "";
+  std::vector<int> device_ids = std::vector<int>();
+  int output_device = -1;
+  std::string backend_name = "";
+  // The DDPLoggingData is logged in which iteration of the training loop,
+  // 0 if the data is logged during DistributedDataParallel construction time.
+  int iteration = -1;
+  // Parameter's data type
+  std::string dtype = "";
+  // Total parameters size (Bytes)
+  int parameter_size = -1;
+  // The number of parameter tensors
+  int num_parameters = -1;
+  // A list of bucket sizes (Bytes) calculated during construction time
+  std::vector<int> bucket_sizes = std::vector<int>();
```

Comment:
right, I changed it back from string to vector<int>, because DDPLoggingData is a generic struct that can be used by different logging APIs depending on users' applications.  vector<int> gives better flexibility for users to either log them as list or concatenate them as string in logging API level (I concatenated them in our logging APIs)